src/main/java/org/folio/repository/titles/TitlesRepositoryImpl.java

```diff
@@ -61,7 +77,102 @@ public TitlesRepositoryImpl(Vertx vertx) {
     return mapVertxFuture(future).thenApply(result -> null);
   }
 
+
+  @Override
+  public CompletableFuture<List<DbTitle>> getTitlesByResourceTags(List<String> tags, int page, int count, String tenant) {
+    int offset = (page - 1) * count;
+
+    JsonArray parameters = new JsonArray();
+    tags.forEach(parameters::add);
+    parameters
+      .add(offset)
+      .add(count);
+
+    final String query = String.format(SELECT_TITLES_BY_RESOURCE_TAGS,
+      getResourcesTableName(tenant), getTagsTableName(tenant), getHoldingsTableName(tenant), createPlaceholders(tags.size()));
+
+    PostgresClient postgresClient = PostgresClient.getInstance(vertx, tenant);
+
+    LOG.info("Select titles by resource tags = " + query);
+    Future<ResultSet> future = Future.future();
+    postgresClient.select(query, parameters, future.completer());
+
+    return mapResult(future, this::mapTitles);
+  }
+
+  @Override
+  public CompletableFuture<Integer> countTitlesByResourceTags(List<String> tags, String tenant) {
+    JsonArray parameters = new JsonArray();
+    tags.forEach(parameters::add);
+    final String query = String.format(COUNT_TITLES_BY_RESOURCE_TAGS, getTagsTableName(tenant), createPlaceholders(tags.size()));
+
+    PostgresClient postgresClient = PostgresClient.getInstance(vertx, tenant);
+
+    LOG.info("Select packages by tags = " + query);
+    Future<ResultSet> future = Future.future();
+    postgresClient.select(query, parameters, future.completer());
+
+    return mapResult(future, this::readTagCount);
+  }
+
+  private String createPlaceholders(int size) {
+    return String.join(",", Collections.nCopies(size, "?"));
+  }
+
+  private Integer readTagCount(ResultSet resultSet) {
+    return resultSet.getRows().get(0).getInteger("count");
+  }
+
+  private List<DbTitle> mapTitles(ResultSet resultSet) {
+    return mapItems(resultSet.getRows(), this::mapTitle);
+  }
+
+  private DbTitle mapTitle(JsonObject row) {
+    Title title = readHolding(row)
+      .map(this::mapHoldingToTitle)
+      .orElse(null);
+    return DbTitle.builder()
+      .id(Long.parseLong(row.getString("id")))
+      .name(row.getString(ResourceTableConstants.NAME_COLUMN))
+      .title(title)
+      .build();
+  }
+
+  private Optional<Holding> readHolding(JsonObject row){
+    try {
+      if(row.getString("holding") != null) {
+        return Optional.of(ObjectMapperTool.getMapper().readValue(row.getString("holding"), Holding.class));
+      }
+      else{
+        return Optional.empty();
+      }
+    } catch (IOException e) {
+      return Optional.empty();
+    }
+  }
+
+  private Title mapHoldingToTitle(Holding holding) {
+    return Title.builder()
+      .titleName(holding.getPublicationTitle())
+      .titleId(Integer.parseInt(holding.getTitleId()))
+      .pubType(holding.getResourceType())
+      .publisherName(holding.getPublisherName())
+      .build();
```

Comment:
Since we are really using only a handful of entries from the holdings table, does it make sense to pre-process those entries and load only these into the DB table to reduce insertion time into DB or will pre-processing take more time than actual insertion into DB?

Reply:
Yes we can only store fields that we need from /holdings, 
will update this PR with these changes

Reply:
ty @andrii-paias